### PR TITLE
babybear ntt bench, added on-device option

### PR DESCRIPTION
**Pull request by vhnatyk**
## Describe the changes

This PR 

- adds babybear Rust ntt benches
- adds on-device option to get more meaningful stats excluding data transfer
- more flexible launching by introducing bench range

## Linked Issues

Resolves #

### CODE REVIEW

File: wrappers/rust/icicle-core/src/ntt/mod.rs [hunk 1]
```diff
@@ -422,20 +422,41 @@ macro_rules! impl_ntt_bench {
...
+        fn get_min_max_log_size(min_log2_default: u32, max_log2_default: u32) -> (u32, u32) {
+
+            fn get_env_log2(key: &str, default: u32) -> u32 {
+                env::var(key).unwrap_or_else(|_| default.to_string()).parse().unwrap_or(default)
+            }
+
+            let min_log2 = get_env_log2("MIN_LOG2", min_log2_default);
+            let max_log2 = get_env_log2("MAX_LOG2", max_log2_default);
+
+            assert!(min_log2 >= min_log2_default);
```
Comment (LeonHibnik) [inline wrappers/rust/icicle-core/src/ntt/mod.rs:454]: Why limiting here?

Comment (LeonHibnik) [reply wrappers/rust/icicle-core/src/ntt/mod.rs:454]: It that case let's skip these benches or edit the error message to be more clear

File: wrappers/rust/icicle-core/src/ntt/tests.rs [hunk 2]
```diff
@@ -331,6 +331,9 @@ where
                         config.ordering = ordering;
                         let mut batch_ntt_result = vec![F::zero(); batch_size * test_size];
                         for alg in [NttAlgorithm::Radix2, NttAlgorithm::MixedRadix] {
+                            if alg == NttAlgorithm::Radix2 && ordering as u32 > 3 {
+                                continue;
```
Comment (LeonHibnik) [inline wrappers/rust/icicle-core/src/ntt/tests.rs:335]: Add a comment on why we are skipping these runs

File: wrappers/rust/icicle-core/src/ntt/mod.rs (hunk 1 above)
Comment (vhnatyk) [reply wrappers/rust/icicle-core/src/ntt/mod.rs]: added error messages

File: wrappers/rust/icicle-core/src/ntt/tests.rs (hunk 2 above)
Comment (vhnatyk) [reply wrappers/rust/icicle-core/src/ntt/tests.rs:336]: added comment

**Review by LeonHibnik — APPROVED** [review]

**Pull request merged**